src/middleware/withAPIPageLogger.ts

```diff
@@ -0,0 +1,56 @@
+import { createLogger } from "@fewlines/fwl-logging";
+import { NextApiRequest, NextApiResponse } from "next";
+import { Handler } from "src/pages/api/identity";
+
+type WithAPIPageLogger = (
+  request: NextApiRequest,
+  response: NextApiResponse,
+) => Promise<JSON | void>;
+
+export function withAPIPageLogger(handler: Handler): WithAPIPageLogger {
+  const logger = createLogger("connect-account", "json").withMeta({
+    process: "getServerSideProps",
+  });
+
+  const processTimeStart = process.hrtime.bigint();
+
+  return async (request, response) => {
+    handler(request, response)
```

Comment:
Thinking out loud: we could have a problem there with passing `response`, inside this function, we could do `response.json(thing)` and it would result in an error of the type "response has already been sent".

Do you think it would be alright not to pass `response`?

Reply:
I see the issue (and got it during one of my first iteration for this task if I remember correctly).

I not sure about how to handle it this way thought. Would it means that we need to do something like `export default withAPIPageLogger(handler).then(({data}) => response.json({data})).catch(...);` ? Seems off.

Reply:
Well, I would try something like:

```ts
interface HandlerReponse {
  status: HttpStatus;
  data?: unknown;
  headers?: Record<string, string>;
}
type Handler = (request: NextApiRequest) => Promise<HandlerResponse>
```

I'm not sure it covers all posibilities but I assume it fixes the http 200 problem along with the problem

Reply:
`withAPIPageLogger` shouldn't return a promise, right?

Reply:
Well, it can be either `void` or `Promise<void>` I guess, since you can export an async function